pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -71,24 +82,71 @@ var NewReconciler = jobframework.NewGenericReconciler(
 		return &Pod{}
 	}, nil)
 
-type Pod corev1.Pod
+type Pod struct {
+	pod     corev1.Pod
+	isGroup bool
+	list    corev1.PodList
+	isFull  bool // if it's fully read
+}
 
-var _ jobframework.GenericJob = (*Pod)(nil)
-var _ jobframework.JobWithCustomStop = (*Pod)(nil)
-var _ jobframework.JobWithFinalize = (*Pod)(nil)
+// Get implements jobframework.ComposableJob.
+func (p *Pod) Get(ctx context.Context, c client.Client, key types.NamespacedName) (err error, removeWlFinalizer bool) {
+	if err := c.Get(ctx, key, &p.pod); err != nil {
+		return err, apierrors.IsNotFound(err)
+	}
+	groupName := p.groupName()
+	p.isGroup = groupName != ""
+	if !p.isGroup {
+		return nil, !p.pod.DeletionTimestamp.IsZero()
+	}
+
+	if err := c.List(ctx, &p.list, client.MatchingLabels{
+		GroupNameLabel: groupName,
+	}, client.InNamespace(key.Namespace)); err != nil {
+		return err, false
+	}
+
+	p.isFull = true
+	groupTotalCount, err := p.groupTotalCount()
+	if err != nil {
+		return err, false
+	}
+
+	if len(p.list.Items) != groupTotalCount {
```

Comment:
This definitely won't match if there are Failed pods that can be replaced.

Reply:
Changed the behavior of Load in 4c035abe